lib/queue.rb

```diff
@@ -1,16 +1,47 @@
 class Queue
 
+  MAX_SIZE = 20
   def initialize
-    # @store = ...
-    raise NotImplementedError, "Not yet implemented"
+    @queue = []
+    @size = 20
+    @front = -1 
+    @rear = -1
+    
   end
 
   def enqueue(element)
-    raise NotImplementedError, "Not yet implemented"
+
+    if (@front == 0 && @rear == @size - 1) 
+    raise "queue is full"
+    elsif (@front == -1)
+      @front = @rear = 0
+      @queue[@rear] = element
+    else
+      @rear += 1
+      @queue[@rear] = element
```

Comment:
This lets the rear wrap around

```suggestion
      @rear = (@rear + 1) % @store.length
      @queue[@rear] = element
```